storage/2020-08-04/queue/queues/delete.go

```diff
@@ -2,84 +2,49 @@ package queues
 
 import (
 	"context"
+	"fmt"
 	"net/http"
 	"strings"
 
-	"github.com/Azure/go-autorest/autorest"
-	"github.com/Azure/go-autorest/autorest/azure"
-	"github.com/Azure/go-autorest/autorest/validation"
-	"github.com/tombuildsstuff/giovanni/storage/internal/endpoints"
+	"github.com/hashicorp/go-azure-sdk/sdk/client"
 )
 
+type DeleteResponse struct {
+	HttpResponse *client.Response
+}
+
 // Delete deletes the specified Queue within the specified Storage Account
-func (client Client) Delete(ctx context.Context, accountName, queueName string) (result autorest.Response, err error) {
-	if accountName == "" {
-		return result, validation.NewError("queues.Client", "Delete", "`accountName` cannot be an empty string.")
-	}
+func (c Client) Delete(ctx context.Context, queueName string) (resp DeleteResponse, err error) {
+
 	if queueName == "" {
-		return result, validation.NewError("queues.Client", "Delete", "`queueName` cannot be an empty string.")
+		return resp, fmt.Errorf("`queueName` cannot be an empty string")
 	}
+
 	if strings.ToLower(queueName) != queueName {
-		return result, validation.NewError("queues.Client", "Delete", "`queueName` must be a lower-cased string.")
+		return resp, fmt.Errorf("`queueName` must be a lower-cased string")
 	}
 
-	req, err := client.DeletePreparer(ctx, accountName, queueName)
-	if err != nil {
-		err = autorest.NewErrorWithError(err, "queues.Client", "Delete", nil, "Failure preparing request")
-		return
+	opts := client.RequestOptions{
+		ContentType: "application/xml; charset=utf-8",
+		ExpectedStatusCodes: []int{
+			http.StatusNoContent,
+		},
+		HttpMethod:    http.MethodDelete,
+		OptionsObject: nil,
+		Path:          queueName,
```

Comment:
can we ensure there's a `/` at the start of this

```suggestion
		Path:          fmt.Sprintf("/%s", queueName),
```